test/client.js

```diff
@@ -315,6 +315,64 @@ test('basic get with query params partially in path', (t) => {
   })
 })
 
+test('basic get returns 400 when configured to throw on errors (callback)', (t) => {
+  t.plan(2)
+
+  const server = createServer((req, res) => {
+    res.statusCode = 400
+    res.end('hello')
+  })
+  t.teardown(server.close.bind(server))
+
+  server.listen(0, () => {
+    const client = new Client(`http://localhost:${server.address().port}`, {
+      keepAliveTimeout: 300e3
+    })
+    t.teardown(client.close.bind(client))
+
+    const signal = new EE()
+    client.request({
+      signal,
+      path: '/',
+      method: 'GET',
+      throwOnError: true
```

Comment:
We need a better name for the option, in this example is not throwing at all.

Reply:
Any ideas? What aspects would you like emphasized? Axios uses `validateStatus` for similar config, got uses `throwHttpErrors`.

Reply:
@ronag any preference on the name?